addons/google_calendar/models/google_sync.py

```diff
@@ -84,6 +85,14 @@ def create(self, vals_list):
                 vals.update({'need_sync': False})
         records = super().create(vals_list)
 
+        # Edge case: when creating 'All Day' recurrent event, the first event is wrongly
+        # synchronized as a single event and then its recurrence creates a duplicated event.
+        if vals_list and isinstance(self, Meeting):
+            forbid_sync = all(not vals.get('need_sync', True) for vals in vals_list)
+            records_to_skip = records.filtered(lambda r: r.need_sync and r.allday and r.recurrency and not r.recurrence_id)
+            if forbid_sync and records_to_skip:
+                records_to_skip.with_context(send_updates=False).need_sync = False
+
```

Comment:
Why not have it in calendar.py ? it would prevent the `if isinstance(self, Meeting)`. What do you think ?